src/cpu.c

```diff
@@ -38,13 +38,11 @@ __attribute((always_inline)) inline static void clear_flag_register(void) {
 
 __attribute__((always_inline)) inline static void LD_REG_REG(uint8_t *register_one, uint8_t register_two) {
     *register_one = register_two;
-    ++cpu.cycle_count;
 }
 
-void LD_8(uint8_t *addr) {
-    *addr = mmu_get_byte(cpu.PC);
-    cpu.PC++;
-    cpu.cycle_count += 2;
+void LD_REG_d8(uint8_t *addr) {
+    uint8_t immediate = mmu_get_byte(cpu.PC + 1);
+    LD_REG_REG(addr, immediate);
```

Comment:
```suggestion
static void LD_REG_d8(uint8_t *reg_addr) {
    uint8_t immediate = mmu_get_byte(cpu.PC + 1);
    LD_REG_REG(reg_addr, immediate);
```

Just to be more explicit about what kind of address.